crates/sui/src/client_commands.rs

```diff
@@ -469,13 +469,21 @@ impl SuiClientCommands {
                 let build_config =
                     resolve_lock_file_path(build_config, Some(package_path.clone()))?;
 
-                let compiled_package = build_move_package(
-                    &package_path,
-                    BuildConfig {
-                        config: build_config,
-                        run_bytecode_verifier: true,
-                        print_diags_to_stderr: true,
-                    },
+                let resolution_graph = build_config
+                    .resolution_graph_for_package(&package_path, &mut std::io::stderr())
+                    .map_err(|err| SuiError::ModuleBuildFailure {
+                        error: format!("{:?}", err),
+                    })?;
+
+                if !with_unpublished_dependencies {
+                    ensure_published_dependencies(&resolution_graph)?;
```

Comment:
Heads up (can be resolved in a follow-up PR) when `with_unpublished_dependencies` is provided, we will still need to gather the `published-at` addresses of the packages that have them, we just won't be worried if a package doesn't have one (we will assume that it is going to published with this package).

In the latter case, we may also want to check that a dependency without a `published-at` address truly isn't published (i.e. has the self-addresses of its modules set to `0x0`) otherwise that will also be a problem (something to think about for later).